server/src/auth/auth0.js

```diff
@@ -0,0 +1,108 @@
+'use strict';
+
+const auth_utils = require('./utils');
+const logger = require('../logger');
+
+const https = require('https');
+const Joi = require('joi');
+const querystring = require('querystring');
+const url = require('url');
+
+const options_schema = Joi.object().keys({
+  path: Joi.string().required(),
+  id: Joi.string().required(),
+  secret: Joi.string().required(),
+  redirect_url: Joi.string().required(),
+  host: Joi.string().required()
```

Comment:
The other auth providers discern the host from the request coming in - is there any reason to force it to be configured?